action/protocol/staking/handlers.go

```diff
@@ -175,6 +175,10 @@ func (p *Protocol) handleUnstake(ctx context.Context, act *action.Unstake, csm C
 			failureStatus: iotextypes.ReceiptStatus_ErrUnstakeBeforeMaturity,
 		}
 	}
+	if rErr := validateBucketEndorsement(NewEndorsementStateManager(csm.SM()), bucket, false, blkCtx.BlockHeight); rErr != nil {
+		return log, rErr
```

Comment:
in func validateBucketEndorsement(
```
case errors.Is(err, state.ErrStateNotExist):
		if isEndorsed {
			return &handleError{
				err:           errors.New("bucket is not endorsed"),
				failureStatus: iotextypes.ReceiptStatus_ErrInvalidBucketType,
			}
		}
```
should return
```
				err:           errors.New("bucket is not an endorse bucket"),
				failureStatus: iotextypes.ReceiptStatus_ErrInvalidBucketIndex,
```
since the bucket is not an endorse bucket

Reply:
and then if it returns this error, it should be fine to continue?

Reply:
returning error means validation failed, should not continue

Reply:
ok, fix the error return